File: CW/Lista_2/Z5.ml
let rec initSegment x y =
  match (x, y) with
  | ([], _) -> true
  | (hd :: tl, []) -> false
  | (h1 :: t1, h2 :: t2) when h1 = h2 -> initSegment t1 t2
  | _ -> false
;;

File: CW/Lista_2/Z5.scala
def initSegment[A](x: List[A], y: List[A]): Boolean =
    (x, y) match
        case (Nil, _) => true
        case (head :: tail, Nil) => false
        case (headA :: tailA, headB :: tailB) if headA == headB => initSegment(tailA, tailB) 
        case _ => false

@main def main() =
    println(initSegment(List(), List()))
    println(initSegment(List(), List(1, 2, 3)))
    println(initSegment(List(1, 2), List(1, 2, 3)))
    println(initSegment(List(1, 2), List(1, 3, 2)))
    println(initSegment(List(1, 2), List()))
    